### PR TITLE
Feature/styles

### LINKED ISSUE #11: Add MDBootstrap to the app
Now, our app is working properly we need to make it look better. I'd like you to use MDBootstrap to do so.

https://mdbootstrap.com/

### PR DESCRIPTION
**Pull request by grzegorzroczniak555**
Closes #11

### CODE REVIEW

File: src/app/app.component.scss [hunk 1]
```diff
@@ -0,0 +1,7 @@
+h1,a {
+    margin: 10px 10px 10px 10px;
+}
+
+.btn-no-waves .btn.waves-effect {
+    position: static;
```
Comment (rbobko93) [inline src/app/app.component.scss]: Why did you add those styles? That's weird

File: src/app/app.component.scss [hunk 2]
```diff
@@ -0,0 +1,7 @@
+h1,a {
+    margin: 10px 10px 10px 10px;
```
Comment (rbobko93) [inline src/app/app.component.scss]: You can use bootstrap margin classes instead. For example m-1, m-2 etc.
Also you could have just written: margin: 10px;

File: src/app/task/done-tasks/done-tasks.component.scss [hunk 3]
```diff
@@ -0,0 +1,13 @@
+h2 {
+    margin: 10px 10px 10px 10px;
```
Comment (rbobko93) [inline src/app/task/done-tasks/done-tasks.component.scss]: Same stuff with margins

File: src/app/task/done-tasks/done-tasks.component.scss [hunk 4]
```diff
@@ -0,0 +1,13 @@
+h2 {
```
Comment (rbobko93) [inline src/app/task/done-tasks/done-tasks.component.scss]: Also try not to add styles based on html tag. instead create specific class and then add it to the element. This way if you ever add new h2 elements to the view they won't automatically get those styles. :)

File: src/app/task/done-tasks/done-tasks.component.scss [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+    color: darkblue;
+}
+
+li {
+    margin: 0px 0px 0px 10px;
+}
+
+.list-group-item {
+    border-color: black;
+    width: max-content;
```
Comment (rbobko93) [inline src/app/task/done-tasks/done-tasks.component.scss]: You can use bootstrap class w-100 which makes element take 100% width of its parent

File: src/app/task/task.component.html [hunk 6]
```diff
@@ -1,25 +1,28 @@
...
-        <label> Name:
-      <input type="text" formControlName="name">
-    </label>
+        <label for="form1" class=''> Name:
+            <input mdbInput type="text" id="form1" class="form-control" formControlName="name">
+        </label>
         <label> Description:
-      <input type="text" formControlName="description">
-    </label>
+            <input mdbInput type="text" id="form1" class="form-control" formControlName="description">
```
Comment (rbobko93) [inline src/app/task/task.component.html]: id of the element should indicate what it is used for. For example for this one it should be description or task-description

File: src/app/task/task.component.html [hunk 7]
```diff
@@ -1,25 +1,28 @@
 <h2>Task To Do: </h2>
 
 <div>
-
     <form [formGroup]="taskForm" (ngSubmit)="addTask()">
-        <label> Name:
-      <input type="text" formControlName="name">
-    </label>
+        <label for="form1" class=''> Name:
+            <input mdbInput type="text" id="form1" class="form-control" formControlName="name">
```
Comment (rbobko93) [inline src/app/task/task.component.html]: You have used from1 id twice. Id's on elements should be unique throughout whole template.

File: src/app/task/task.component.html [hunk 8]
```diff
@@ -1,25 +1,28 @@
 <h2>Task To Do: </h2>
 
 <div>
-
     <form [formGroup]="taskForm" (ngSubmit)="addTask()">
-        <label> Name:
-      <input type="text" formControlName="name">
-    </label>
+        <label for="form1" class=''> Name:
```
Comment (rbobko93) [inline src/app/task/task.component.html]: Empty class attr? Remove it then

File: src/app/task/task.component.html [hunk 9]
```diff
@@ -1,25 +1,28 @@
...
         <label> Description:
-      <input type="text" formControlName="description">
-    </label>
+            <input mdbInput type="text" id="form1" class="form-control" formControlName="description">
+        </label>
+
 
-        <button type="submit">Add Task</button>
+        <button type="submit" mdbBtn mdbWavesEffect color="primary" title="Adding a task">
+            <p class="add-task">Add Task</p>
```
Comment (rbobko93) [inline src/app/task/task.component.html]: You don't have to add button text in p tags. If you want to style it, wrap it in span. p is meant for paragraphs. span is for this kind of custom work

Comment (rbobko93) [reply src/app/task/task.component.html]: https://stackoverflow.com/questions/1908234/when-to-use-span-instead-p

File: src/app/task/task.component.scss [hunk 10]
```diff
@@ -0,0 +1,28 @@
+* {
```
Comment (rbobko93) [inline src/app/task/task.component.scss]: You cannot do anything like this. * is unacceptable and you should remove it.

File: src/app/task/task.component.scss [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+h2 {
+    color: darkblue;
+}
+
+button {
+    padding: 0px 0px 0px 0px;
+}
+
+.add-task:hover {
```
Comment (rbobko93) [inline src/app/task/task.component.scss]: I am not sure if bootstrap has any classes for this kind of styling but before writing your own styles you should always checkout if bootstrap already has it.

File: src/app/app.component.scss (hunk 1 above)
Comment (grzegorzroczniak555) [reply src/app/app.component.scss]: It was added to one of parts of mdb as a style, but in fact it doesn't change anything :D I've deleted

File: src/app/app.component.scss (hunk 2 above)
Comment (grzegorzroczniak555) [reply src/app/app.component.scss]: image

I've done it. I am wondering - Can I use it in this way and set style for a m-1 class? Because for now I've done it using separate class from mdb for h1 and p tags (font-weight-bold and text-warning)

Comment (grzegorzroczniak555) [reply src/app/app.component.scss]: I already know - I could use for example :first-child

File: src/app/task/task.component.scss (hunk 11 above)
Comment (grzegorzroczniak555) [reply src/app/task/task.component.scss]: https://mdbootstrap.com/docs/jquery/utilities/spacing/ - there is something like it - I'll remember about your suggestion.

File: src/app/task/done-tasks/done-tasks.component.scss (hunk 5 above)
Comment (grzegorzroczniak555) [reply src/app/task/done-tasks/done-tasks.component.scss]: image
How to create a multiple class? It doesn't work in this way.. when I put it into <div>, it doesn't help

File: src/app/app.component.scss (hunk 2 above)
Comment (rbobko93) [reply src/app/app.component.scss]: I don't know what you were talking about exactly.

File: src/app/task/done-tasks/done-tasks.component.scss (hunk 5 above)
Comment (rbobko93) [reply src/app/task/done-tasks/done-tasks.component.scss]: create multiple class? What does that mean? Also, you don't have to add screenshots here. After you push you can reference specific line from file and it will be displayed here. 
https://stackoverflow.com/questions/23821235/how-to-link-to-specific-line-number-on-github

If you want to add multiple classes to html tag you would do: class="class1 class2 class3"

File: src/app/app.component.html [hunk 12]
```diff
@@ -1,8 +1,8 @@
-<span class="tasks">
-   <a><img src = "task/task.jpg" alt="" height="20" weight="20"></a>
+<img src="task/task.jpg" alt="xxx" height="20" width="20">
```
Comment (rbobko93) [inline src/app/app.component.html]: If you are adding alt don't give it this kind of message. Like I've said it actually has a proper use case.

File: src/app/app.component.scss [hunk 13]
```diff
@@ -1,7 +1,4 @@
-h1,a {
-    margin: 10px 10px 10px 10px;
+.font-weight-bold,
+.text-warning {
```
Comment (rbobko93) [inline src/app/app.component.scss]: Still could use margin classes from bootstrap

File: src/app/task/done-tasks/done-tasks.component.html [hunk 14]
```diff
@@ -1,12 +1,14 @@
-<h2> Done Tasks </h2>
+<h2 class="done-task-title"> Done Tasks </h2>
```
Comment (rbobko93) [inline src/app/task/done-tasks/done-tasks.component.html]: Classes should be named by what they change. The name you used could be used as id not class name. For example if class changes text color to red, name would be text-red.

File: src/app/app.component.scss (hunk 13 above)
Comment (rbobko93) [reply src/app/app.component.scss]: Also if those classes are from bootstrap, do not change them. Add a new class, your own and add it to the html element.

File: src/app/task/task.component.html [hunk 15]
```diff
@@ -1,25 +1,25 @@
-<h2>Task To Do: </h2>
+<h2 class="task-to-do-title">Task To Do: </h2>
```
Comment (rbobko93) [inline src/app/task/task.component.html]: Same thing here. If you are adding element specific styles, add id to it and add styles using this id.

File: src/app/task/task.component.html [hunk 16]
```diff
@@ -1,25 +1,25 @@
...
 
 <div>
-    <form [formGroup]="taskForm" (ngSubmit)="addTask()">
-        <label for="form1" class=''> Name:
-            <input mdbInput type="text" id="form1" class="form-control" formControlName="name">
+    <form [formGroup]="taskForm" (ngSubmit)="addTask()" class="forms">
+        <label for="form"> Name:
+            <input mdbInput type="text" id="task-form" class="form-control" formControlName="name">
```
Comment (rbobko93) [inline src/app/task/task.component.html]: id="task-name"

File: src/app/task/task.component.html [hunk 17]
```diff
@@ -1,25 +1,25 @@
...
         </label>
         <label> Description:
-            <input mdbInput type="text" id="form1" class="form-control" formControlName="description">
+            <input mdbInput type="text" id="task-description-form" class="form-control" formControlName="description">
```
Comment (rbobko93) [inline src/app/task/task.component.html]: id="task-description"
